===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(swHashMap_new:root)) at 22 (line 115, column 10) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/swoole-src/src/core/hashmap.c
+++ b/table1/benchmarks/swoole-src/src/core/hashmap.c
@@ -114,6 +114,7 @@ swHashMap* swHashMap_new(uint32_t bucket_num, swHashMap_dtor dtor)
     root->hh.tbl = (UT_hash_table*) sw_malloc(sizeof(UT_hash_table));
     if (!(root->hh.tbl))
     {
+        free(root);
         swWarn("malloc for table failed.");
         return NULL;
     }
